externs/browser/webgl2_compute.js

```diff
@@ -0,0 +1,1608 @@
+WebGL2ComputeRenderingContext.prototype.TEXTURE_ALPHA_TYPE;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.TEXTURE_DEPTH_TYPE;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.TEXTURE_COMPRESSED;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.SAMPLER_2D_MULTISAMPLE;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.INT_SAMPLER_2D_MULTISAMPLE;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.UNSIGNED_INT_SAMPLER_2D_MULTISAMPLE;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.VERTEX_ATTRIB_BINDING;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.VERTEX_ATTRIB_RELATIVE_OFFSET;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.VERTEX_BINDING_DIVISOR;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.VERTEX_BINDING_OFFSET;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.VERTEX_BINDING_STRIDE;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.VERTEX_BINDING_BUFFER;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.MAX_VERTEX_ATTRIB_RELATIVE_OFFSET;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.MAX_VERTEX_ATTRIB_BINDINGS;
+
+/** @const {number} */
+WebGL2ComputeRenderingContext.prototype.MAX_VERTEX_ATTRIB_STRIDE;
+
+
+/**
+ * @param {number} num_groups_x
+ * @param {number} num_groups_y
+ * @param {number} num_groups_z
+ * @return {undefined}
+ */
+WebGL2ComputeRenderingContext.prototype.dispatchCompute = function(
+    num_groups_x, num_groups_y, num_groups_z) {};
+
+/**
+ * @param {number} offset
+ * @return {undefined}
+ */
+WebGL2ComputeRenderingContext.prototype.dispatchComputeIndirect = function(
+    offset) {};
+
+/**
+ * @param {number} mode
+ * @param {number} offset
+ * @return {undefined}
+ */
+WebGL2ComputeRenderingContext.prototype.drawArraysIndirect = function(
+    mode, offset) {};
+
+/**
+ * @param {number} mode
+ * @param {number} type
+ * @param {number} offset
+ * @return {undefined}
+ */
+WebGL2ComputeRenderingContext.prototype.drawElementsIndirect = function(
+    mode, type, offset) {};
+
+/**
+ * @param {number} target
+ * @param {number} pname
+ * @param {number} param
+ * @return {undefined}
+ */
+WebGL2ComputeRenderingContext.prototype.framebufferParameter = function(
+    target, pname, param) {};
+
+/**
+ * @param {number} target
+ * @param {number} pname
+ * @return {*}
+ */
+WebGL2ComputeRenderingContext.prototype.getFramebufferParameter = function(
+    target, pname) {};
+
+/**
+ * @param {!WebGLProgram} program
+ * @param {number} programInterface
+ * @param {number} pname
+ * @return {*}
+ */
+WebGL2ComputeRenderingContext.prototype.getProgramInterfaceParameter = function(
+    program, programInterface, pname) {};
+
+/**
+ * @param {!WebGLProgram} program
+ * @param {number} programInterface
+ * @param {number} index
+ * @param {!Array<number>} props
+ * @return {?Array<*>}
```

Comment:
The return type doesn't seem to match the spec.

`any getProgramResource(WebGLProgram program, GLenum programInterface, GLuint index, GLenum props);`

Reply:
The right signature should be `sequence? getProgramResource(WebGLProgram program, GLenum programInterface, GLuint index, sequence props)` as described in https://www.khronos.org/registry/webgl/specs/latest/2.0-compute/#program-interface. We will fix the typo in the spec later. However, I'm not quite sure if `{?Array<*>}` is valid here.

Reply:
It looks correct to me given https://www.w3.org/TR/WebIDL-1/#es-sequence, although I think `?Array<?>` would be equivalent in practice. Unless there's something I'm missing.